netmiko/adva/adva_aos_fsp_150_f3.py

```diff
@@ -0,0 +1,207 @@
+            cmd_verify (bool, optional): Verify command echo before proceeding. Defaults to True.
+            pattern (Optional[str], optional): Regular expression pattern used to identify that reading is done. Defaults to None.
+
+        Returns:
+            str: SSH Channel Output
+        """
+        # Multiline Command Below is used for paging
+        command = (
+            "configure user-security"
+            + self.TELNET_RETURN
+            + "config-user "
+            + self.username
+            + " cli-paging disabled"
+            + self.TELNET_RETURN
+            + "home"
+            + self.TELNET_RETURN
+        )
+        return super().disable_paging(
+            command=command,
+            delay_factor=delay_factor,
+            cmd_verify=cmd_verify,
+            pattern=pattern,
+        )
+
+    def set_base_prompt(
+        self,
+        pri_prompt_terminator: str = "#",
+        alt_prompt_terminator: str = ">",
+        delay_factor: float = 1.0,
+        pattern: Optional[str] = None,
+    ) -> str:
+        """Sets self.base_prompt
+
+        Used as delimiter for stripping of trailing prompt in output.
+
+        Should be set to something that is general and applies in multiple contexts. For Cisco
+        devices this will be set to router hostname (i.e. prompt without > or #).
+
+        This will be set on entering user exec or privileged exec on Cisco, but not when
+        entering/exiting config mode.
+
+        :param pri_prompt_terminator: Primary trailing delimiter for identifying a device prompt
+
+        :param alt_prompt_terminator: Alternate trailing delimiter for identifying a device prompt
+
+        :param delay_factor: See __init__: global_delay_factor
+
+        :param pattern: Regular expression pattern to search for in find_prompt() call
+        """
+        prompt = self.find_prompt()
+        match = re.search(r"(^.+?)-->$", prompt)
+        if not match:
+            raise ValueError("Router prompt not found: {0}".format(repr(prompt)))
+        self.base_prompt = match[1]
+        return self.base_prompt
+
+    def send_config_set(
+        self,
+        config_commands: Union[str, Sequence[str], TextIO, None] = None,
+        exit_config_mode: bool = True,
+        read_timeout: Optional[float] = 2.0,
+        delay_factor: Optional[float] = None,
+        max_loops: Optional[int] = None,
+        strip_prompt: bool = False,
+        strip_command: bool = False,
+        config_mode_command: Optional[str] = None,
+        cmd_verify: bool = True,
+        enter_config_mode: bool = True,
+        error_pattern: str = "",
+        terminator: str = r"#",
+        bypass_commands: Optional[
+            str
+        ] = r"(add\s+\w+\s+[A-Za-z0-9#\?!@\$%\^&\*-]*\s+[A-Za-z0-9#\?!@\$%\^&\*-]*\s+(superuser|crypto|maintenance|provisioning|retrieve|test-user)|secret.*)",
```

Comment:
I ran into issues with the Adva CLI for the F3 device obsicating eg hiding passwords for commands such as adding users or tacacs secrets.

When you pass config with plain text passwords as config, the CLI converts the password into *****

test-bally-532824--> configure user-security 
test-bally-532824:user-sec--> add guych **** **** 

This created problems with cmd_verify, so instead of disabling cmd_verify globally I though it made sense use the bypass_commands feature, which I think for the specified commands disables cmd_verify? Which from testing resolves the issue with the echo of the command back not getting what was sent to the channel because the password has been hidden

Reply:
Instead of using that big character-set, can we use \S+ (which means non-whitespace)?

```python
r"(add\s+\w+\s+\S*\s+\S*\s+(superuser|crypto|maintenance|provisioning|retrieve|test-user)|secret.*)"
```

We also should prefix both of these to make them non-capturing so:

```python
r"(?:add\s+\w+\s+\S*\s+\S*\s+(?:superuser|crypto|maintenance|provisioning|retrieve|test-user)|secret.*)"
```
Basically the `?:` means we are using the parenthesis as a logical-or and not for capturing/replaying what we match. There is a case(s) where having capturing patterns can cause breakage in Netmiko. Not sure in matters here, but for safety it is better to use non-capturing (since we don't need it).

Reply:
Ok perfect thanks for the suggestion.

`r"(?:add\s+\S+\s+\S+\s+\S+\s+(?:superuser|crypto|maintenance|provisioning|retrieve|test-user)|secret.*)"`

The above seems to work nicely